doc/why_mf_next.md

```diff
@@ -0,0 +1,116 @@
+# Why `MessageFormat` needs a successor ([issue #49](https://github.com/unicode-org/message-format-wg/issues/49))
+
+## Intro
+
+The `MessageFormat` API and syntax have been around for a long time.
+
+Intro
+* `MessageFormat` is the Unicode API for software localization
+* It is 20 years old, well designed, proven solution
+* Its design is optimized for the software development model of 20y ago and its
+shortcomings result in mixed reception and adoption by the industry.
+
+The current wave of software development uses dynamic languages, modern UI
+frameworks and new forms of user interactions (voice, VR etc.).
+
+Considering these new challenges, combined with the lessons learned from using
+`MessageFormat`, we aim to design the next iteration of `MessageFormat`
+suitable for current generation of software, and adoption by Web Standards.
+
+Other efforts: [Fluent](https://projectfluent.org/),
+[FBT](https://facebook.github.io/fbt/)
+
+## Core problems with the current `MessageFormat`
+
+1. Does not have any “extension points”
+2. Can't remove anything, even if now we know better
+3. Hard to map to the existing localization core structures
+4. Designed to be API only, plain text, UI, “imperative style”
+
+### 1. Does not have any “extension points”
+
+No extension points means that it is hard to add new functionality unless you
+are doing it in ICU itself.
+It also means most tools used to process these messages are built rigidly,
+and are unprepared to handle changes
+(think localization tools, liners, friendly UIs, etc.).
```

Comment:
I'd maybe add that, from the experience with Fluent, the fact that almost every feature in MF is a syntax feature, means that on the very lowest level - AST, parser - every extensions basically breaks the existing functionality because the file cannot be parsed and its AST operated on.

Fluent's AST is more generic which means that more of the functionality falls on higher levels of abstraction and in fact, over the last 3 years, all accrued "wants" can be implemented either on the higher level (Semantic Comments, Dynamic References etc.) or require additive syntax relaxations (flatten selectors, rich overlays).